### PR TITLE
Make ERRORPRONE warnings into errors by changing javacopts

### PR DESCRIPTION
**Pull request by petroseskinder**
based off of @greggdonovan's suggestions in #15, this should help with checking for any Java/BUILD formatting and unused deps issues.

### CODE REVIEW

File: tools/build_rules/java.bzl [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		'-Xep:TruthConstantAsserts:WARN',
+		'-Xep:TypeParameterShadowing:WARN',
+		'-Xep:TypeParameterUnusedInFormals:WARN',
+		'-Xep:URLEqualsHashCode:WARN',
+		'-Xep:UnsynchronizedOverridesSynchronized:WARN',
+		'-Xep:WaitNotInLoop:WARN',
+]
+
+def java_library(javacopts=[], **kwargs):
+	if native.repository_name() == '@':
```
Comment (cgrushko) [inline tools/build_rules/java.bzl:66]: @greggdonovan
nice. I was leaning towards using `.bazelrc` for this, but it doesn't let skip this in external repos.

**Comment by cgrushko** [comment]
retest this please

**Review by cgrushko — APPROVED** [review]

**Comment by cgrushko** [comment]
Thanks!!

**Comment by cgrushko** [comment]
@petroseskinder you probably can't merge because ci.bazel.io is stuck?
The logs from the second ci.bazel.io check indicate the tests are passing, so let's "force" merge this.
If you can't, tell me and I'll do it.

**Comment by petroseskinder** [comment]
> you probably can't merge because ci.bazel.io is stuck?

You are correct. I am currently unable to merge this. Let's shoot for force merging.

**Pull request merged**